devkit/states/vim/pr_vimrc_init

```diff
@@ -1,22 +1,364 @@
+  autocmd FileType go nmap     <silent> <leader>r <Plug>(go-run)
+  autocmd FileType go nmap     <silent> <leader>R <Plug>(go-rename)
+  autocmd FileType go nmap     <silent> <leader>t <Plug>(go-test-func)
+  autocmd FileType go nnoremap <silent> <leader>u :GoFillStruct<CR>
+  autocmd FileType go nnoremap <silent> <leader>y :GoKeyify<CR>
+
+augroup END
+
+"----------------------
+" itchyny/lightline.vim
+"----------------------
+set laststatus=2
+set updatetime=100
+set noshowmode " mode shown in lightline status
+
+let g:lightline = {
+  \ 'active': {
+  \   'left': [ [ 'mode', 'paste', 'readonly' ],
+  \             [ 'gitbranch', 'filename', 'modified' ], [ 'gostatus' ] ]
+  \ },
+  \ 'component_function': {
+  \   'gitbranch': 'fugitive#head',
+  \ },
+  \ 'component': {
+  \   'gostatus': '%#goStatuslineColor#%{LightlineGo()}',
+  \ },
+  \ 'component_type': {
+  \   'gostatus': 'raw',
+  \ },
+  \ }
+
+function! LightlineGo()
+  return exists('*go#statusline#Show') ? go#statusline#Show() : ''
+endfunction
+
+"-----------------
+" junegunn/fzf.vim
+"-----------------
+let g:fzf_layout = { 'down': '~50%' }
+
+" TODO changed some of these, okay or not?
+noremap <leader>zb :BCommits<CR>
+noremap <leader>zc :Commits<CR>
+noremap <leader>zC :Commands<CR>
+noremap <leader>zcl :Colors<CR>
+noremap <leader>zf :Files<CR>
+noremap <leader>zt :Filetypes<CR>
+noremap <leader>zg :GFiles?<CR>
+noremap <leader>zh :History:<CR>
+noremap <leader>zl :Lines<CR>
+noremap <leader>zm :Maps<CR>
+noremap <leader>zs :Snippets<CR>
+
+let g:rg_command = '
+  \ rg --column --line-number --no-heading --fixed-strings --ignore-case --no-ignore --hidden --follow --color "always"
+  \ -g "*.{js,json,php,md,styl,jade,html,config,py,cpp,c,go,hs,rb,conf}"
+  \ -g "!{.git,node_modules,vendor}/*" '
+
+command! -bang -nargs=* Rg
+  \ call fzf#vim#grep(
+  \   'rg --column --line-number --no-heading --color=always '.shellescape(<q-args>), 1,
+  \   <bang>0 ? fzf#vim#with_preview('up:60%')
+  \           : fzf#vim#with_preview('right:50%:hidden', '?'),
+  \   <bang>0)
+
+command! -bang -nargs=* F call fzf#vim#grep(g:rg_command .shellescape(<q-args>), 1, <bang>0)
+
+"------------------
+" majutsushi/tagbar
+"------------------
+let g:tagbar_compact = 1
+let g:tagbar_sort = 0
+let g:tagbar_autofocus = 1
+let g:tagbar_width = 30
+nmap <S-F10> :TagbarToggle<CR>
+
+"--------------------
+" scrooloose/nerdtree
+"--------------------
+let NERDTreeShowBookmarks=1
+let NERDTreeIgnore=['\\.swo$', '\\.swp$', '\\.git']
+let NERDTreeChDirMode=0
+let NERDTreeQuitOnOpen=1
+let NERDTreeShowHidden=1
+let NERDTreeKeepTreeInNewTab=1
+noremap <S-F8> :NERDTreeToggle<CR>
+noremap <leader>nt :NERDTreeFind<CR>
+
+"-----------------
+" SirVer/ultisnips
+"-----------------
+let g:UltiSnipsExpandTrigger="<tab>"
+let g:UltiSnipsJumpForwardTrigger="<c-j>"
+let g:UltiSnipsJumpBackwardTrigger="<c-k>"
+" TODO the plugged directory is for plugins
```

Comment:
There is a .vim/ultisnips directory but the snippets don't get installed there.  We can move them there.  Ultisnips using a different format of snippets, that is why I specified a directory that didn't include non-compatible snippets.

Reply:
the default is ~/.vim/UltiSnips/vim.snippets, any reason not to leave it?